### PR TITLE
Fixed fees calculation

### PR DESCRIPTION
**Pull request by stefanoregis**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please make your selection.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change requires a new dependency
- [ ] Code Maintainability / comments
- [ ] Code Refactoring / future-proofing

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added pytest unit tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pycryptobot.py [hunk 1]
```diff
@@ -88,7 +88,8 @@ def calculateMargin(buy_size: float=0.0, buy_filled: int=0.0, buy_price: int=0.0
     sell_size = (sell_percent / 100) * ((sell_price / buy_price) * buy_size)
 
     if sell_fee == 0.0 and sell_taker_fee > 0.0:
-        sell_fee = sell_size * sell_taker_fee
+        #sell_fee = sell_size * sell_taker_fee
```
Comment (whittlem) [inline pycryptobot.py:91]: I don't think this is correct. Fees are based on the trade size not the price. For example if the Bitcoin price is £40,000 and your trade is £1,000 the fees is calculated on the £1,000 not £40,000.

File: troubleshoot_margin.py [hunk 2]
```diff
@@ -12,7 +12,8 @@ def calculateMargin(buy_size: float=0.0, buy_filled: int=0.0, buy_price: int=0.0
     sell_size = (sell_percent / 100) * ((sell_price / buy_price) * buy_size)
 
     if sell_fee == 0.0 and sell_taker_fee > 0.0:
-        sell_fee = sell_size * sell_taker_fee
+        #sell_fee = sell_size * sell_taker_fee
```
Comment (whittlem) [inline troubleshoot_margin.py:15]: Same as above... the fees is on the trade size not the price.

File: pycryptobot.py (hunk 1 above)
Comment (stefanoregis) [reply pycryptobot.py:91]: yes you are perfectly right i understand what you mean

Comment (whittlem) [reply pycryptobot.py:91]: What is this pull request actually changing now? It looks like you just have undone your previous pull request? Does this update work? Have you confirmed with Mario?

Comment (stefanoregis) [reply pycryptobot.py:91]: This was before we talked, i'm checking it actually

Comment (stefanoregis) [reply pycryptobot.py:91]: I'll make a pull as soon it will be ok

Comment (stefanoregis) [reply pycryptobot.py]: reverted on my second commit, i'm waiting for a chack from Mario as well

File: pycryptobot.py [hunk 3]
```diff
@@ -94,14 +94,15 @@ def calculateMargin(buy_size: float=0.0, buy_filled: int=0.0, buy_price: int=0.0
...
-    buy_value = buy_price + buy_fee
-    sell_value = sell_price - sell_fee
-    profit = sell_value - buy_value
+    
+    if app.getExchange() == 'coinbasepro' : 
+        buy_value = buy_size + buy_fee
+        sell_value = sell_size - sell_fee
+        profit = sell_value - buy_value
+    else :
+        buy_value = buy_price + buy_fee
```
Comment (whittlem) [inline pycryptobot.py:103]: Are we sure this is right for Binance? buy_price + buy_fee? Surely that we should be working off the trade size and not the price. Is maybe the price coming through weirdly for Binance?

Comment (whittlem) [reply pycryptobot.py:103]: I will merge this but I suspect a closer look of Binance is needed.

**Review by whittlem — APPROVED** [review]

File: pycryptobot.py (hunk 3 above)
Comment (stefanoregis) [reply pycryptobot.py:103]: this sure, i still can see some strange stuffs with binance, sometimes the calculated fees are ok, sometimes no

**Pull request merged**